### PR TITLE
Support CodeMirror 6

### PR DESCRIPTION
**Pull request by personalizedrefrigerator**
# Summary

This pull request does the following:
- Upgrades the build script. This allows importing CodeMirror 6 packages. (See [the markdown editor plugin tutorial](https://joplinapp.org/help/api/tutorials/cm6_plugin#update-the-plugin-build-script)).
- Adds support for CodeMirror 6 (the desktop beta markdown editor).

# Notes

- Currently, each command in `commands.js` defines a hint with a `hint` function. In CodeMirror 6, the `hint` function is unused. Instead, the `text` property of the hint is inserted.
- Because changes to `package-lock.json` can be difficult to review, these changes are not included in this pull request.

# Screen recording

https://github.com/hieuthi/joplin-plugin-slash-commands/assets/46334387/8d78ddc5-1411-4c2b-a6ed-af7de9836aa0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,17 +16,18 @@
...
-    "webpack-cli": "^3.3.11",
-    "yargs": "^16.2.0"
+    "copy-webpack-plugin": "^11.0.0",
+    "fs-extra": "^10.1.0",
+    "glob": "^8.0.3",
+    "tar": "^6.1.11",
+    "ts-loader": "^9.3.1",
+    "typescript": "^4.8.2",
+    "webpack": "^5.74.0",
+    "webpack-cli": "^4.10.0"
```
Comment (personalizedrefrigerator) [inline package.json:31]: The new dependency versions in this file were taken from the [upstream `package.json` template](https://github.com/laurent22/joplin/blob/dev/packages/generator-joplin/generators/app/templates/package_TEMPLATE.json).

File: webpack.config.js [hunk 2]
```diff
@@ -6,35 +6,53 @@
 // update, you can easily restore the functionality you've added.
 // -----------------------------------------------------------------------------
```
Comment (personalizedrefrigerator) [inline webpack.config.js:8]: The changes to `webpack.config.js` were taken from the [upstream `webpack.config.js` template](https://github.com/laurent22/joplin/blob/dev/packages/generator-joplin/generators/app/templates/webpack.config.js). This allows CodeMirror 6 packages to be `import`ed by `codeMirror6.ts`.

**Comment by personalizedrefrigerator** [comment]
@hieuthi We've made the CodeMirror 6-based Markdown editor the default in the v3.1 Joplin release. Is this pull request something you would be interested in reviewing? (Thank you for developing this plugin!)

**Comment by hieuthi** [comment]
Yes, but I am a bit busy at the moment. I think I will have some free time around 15 this month.
I will try to make this plugin compatible with mobile around that time.

**Pull request merged**